KiKIMR-22449 release notes 24.3.15.5 (#14069)
ydb-platform/ydb: ydb/docs/en/core/changelog-server.md

## Version 24.3 {#24-3}

### Version 24.3.15.5 {#24-3-15-5}

Release date: February 6, 2025

#### Functionality

* Added the ability to register a [database node](./concepts/glossary.md#database-node) using a certificate. In the [Node Broker](./concepts/glossary.md#node-broker) the flag `AuthorizeByCertificate` has been added to enable certificate-based registration.

> * [Added](https://github.com/ydb-platform/ydb/pull/11775) priorities for authentication ticket through a [third-party IAM provider](./security/authentication#iam), with the highest priority given to requests from new users. Tickets in the cache update their information with a lower priority.

Reviewer: ```suggestion
* [Added](https://github.com/ydb-platform/ydb/pull/11775) priorities for authentication ticket through a [third-party IAM provider](./security/authentication.md#iam), with the highest priority given to requests from new users. Tickets in the cache update their information with a lower priority.
```

## Version 24.3 {#24-3}

### Version 24.3.15.5 {#24-3-15-5}

Release date: February 6, 2025

#### Functionality

> * Added the ability to register a [dynamic node](./concepts/glossary#dynamic) using a certificate. In the [Node Broker](./reference/configuration/#node-broker-config), the system tablet responsible for registering dynamic nodes in the cluster, the flag `AuthorizeByCertificate` has been added to enable certificate-based registration.

Reviewer: ```suggestion
* Added the ability to register a [dynamic node](./concepts/glossary.md#dynamic) using a certificate. In the [Node Broker](./reference/configuration/index.md#node-broker-config), the system tablet responsible for registering dynamic nodes in the cluster, the flag `AuthorizeByCertificate` has been added to enable certificate-based registration.
```
All internal links need to end with `.md`, and for directories, with `index.md`. Otherwise, link correctness checking will not work. More fixes like this are needed below.

Author: Спасибо, в русской версии тоже исправила.

> * Added the ability to register a [dynamic node](./concepts/glossary#dynamic) using a certificate. In the [Node Broker](./reference/configuration/#node-broker-config), the system tablet responsible for registering dynamic nodes in the cluster, the flag `AuthorizeByCertificate` has been added to enable certificate-based registration.

Author: Добавлена Node Broker в глоссарий. Про изменение терминологии - это нужно как-то централизованно менять везде в таком случае, потому что сейчас повсеместно используются статические/динамические узлы.

> * Added the ability to register a [dynamic node](./concepts/glossary#dynamic) using a certificate. In the [Node Broker](./reference/configuration/#node-broker-config), the system tablet responsible for registering dynamic nodes in the cluster, the flag `AuthorizeByCertificate` has been added to enable certificate-based registration.

Reviewer: Она и правда слишком много где используется: по-этому мы в новом контенте стараемся использовать новую, а в старом контенте постепенно по мере сил переезжаем.

> * [Added](https://github.com/ydb-platform/ydb/pull/11775) priorities for authentication ticket through a [third-party IAM provider](./security/authentication#iam), with the highest priority given to requests from new users.

Reviewer: 1. I clicked the link to PR, and it lowers the priority of some requests instead of increasing it.
2. It is not documented in https://ydb.tech/docs/en/security/authentication#iam

Author: Дописала про тикеты с пониженным приоритетом.
Тикет на документацию - https://github.com/ydb-platform/ydb/issues/14305
Предлагаю пропустить пиар без этой доки, мы ее допишем.

> * [Added](https://github.com/ydb-platform/ydb/pull/13748) the ability to configure the coordinator plan resolution using [the dynamically updatable setting](./maintenance/manual/dynamic-config#dynamic-kinds) `immediate_controls_config`.

Reviewer: The link should go to the description of the setting itself, not to the documentation about dynamically updatable settings, as there's not enough information to actually use it. The configuration reference is located at https://ydb.tech/docs/en/reference/configuration/, but this setting hasn't been added yet.

Author: Не понимаю, почему нужно менять ссылку на статическую конфигурацию?

Author: @snaury подскажи, верно ли я написала про твои изменения? Если да, можешь ли ты добавить в https://ydb.tech/docs/en/reference/configuration/ описание настройки `immediate_controls_config`?

Author: Поговорили с Алексеем - считаем, что это настройка, которая нужна разработчикам YDB. В релиз ноуты писать не будем.

#### Performance

* [Improved](https://github.com/ydb-platform/ydb/pull/12747) tablet startup time on large clusters: 210 ms → 125 ms (SSD), 260 ms → 165 ms (HDD).

#### Bug Fixes

> * [Removed](https://github.com/ydb-platform/ydb/pull/11901) the restriction on writing values greater than 127 to the Uint8 type.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12221) an issue where reading small messages from a topic in small chunks significantly increased CPU load. This could lead to delays in reading/writing to the topic.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12915) an issue with restoring from a backup stored in S3 with Path-style addressing.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/13222) an issue with restoring from a backup that was created during an automatic table split.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12601) an issue with Uuid serialization for [CDC](./concepts/cdc).
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12018) an issue with ["frozen" locks](./contributor/datashard-locks-and-change-visibility#vzaimodejstvie-s-raspredelyonnymi-tranzakciyami), which could be caused by bulk operations (e.g., TTL-based deletions).
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12804) an issue where reading from follower of tablets sometimes crashing during automatic table splits.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12807) an issue where the [coordination node](./concepts/datamodel/coordination-node) successfully registered proxy servers despite a connection loss.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/11593) an issue that occurred when opening the interface tab with information about [distributed storage groups](./concepts/glossary#storage-group).
> * [Fixed](https://github.com/ydb-platform/ydb/pull/12448) an issue where Health Check did not report time synchronization issues.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/11658) a rare issue that caused errors during read queries.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/13501) uncommitted changes leak and clean them up on startup.
> * [Fixed](https://github.com/ydb-platform/ydb/pull/13948) consistency issues related to caching deleted ranges.

Reviewer: ```suggestion
* [Removed](https://github.com/ydb-platform/ydb/pull/11901) the restriction on writing values greater than 127 to the Uint8 type.
* [Fixed](https://github.com/ydb-platform/ydb/pull/12221) an issue where reading small messages from a topic in small chunks significantly increased CPU load, which could lead to delays in reading and writing to the topic.
* [Fixed](https://github.com/ydb-platform/ydb/pull/12915) an issue with restoring from a backup stored in S3 with path-style addressing.
* [Fixed](https://github.com/ydb-platform/ydb/pull/13222) an issue with restoring from a backup that was created during an automatic table split.
* [Fixed](https://github.com/ydb-platform/ydb/pull/12601) an issue with Uuid serialization for [CDC](./concepts/cdc.md).
* [Fixed](https://github.com/ydb-platform/ydb/pull/12018) an issue with ["frozen" locks](./contributor/datashard-locks-and-change-visibility.md#interaction-with-distributed-transactions), which could be caused by bulk operations (e.g., TTL-based deletions).
* [Fixed](https://github.com/ydb-platform/ydb/pull/12804) an issue where reading from a follower of tablets sometimes caused crashes during automatic table splits.
* [Fixed](https://github.com/ydb-platform/ydb/pull/12807) an issue where the [coordination node](./concepts/datamodel/coordination-node.md) successfully registered proxy servers despite a connection loss.
* [Fixed](https://github.com/ydb-platform/ydb/pull/11593) an issue that occurred when opening the Embedded UI tab with information about [distributed storage groups](./concepts/glossary.md#storage-group).
* [Fixed](https://github.com/ydb-platform/ydb/pull/12448) an issue where the Health Check did not report time synchronization issues.
* [Fixed](https://github.com/ydb-platform/ydb/pull/11658) a rare issue that caused errors during read queries.
* [Fixed](https://github.com/ydb-platform/ydb/pull/13501) an uncommitted changes leak and cleaned them up on startup.
* [Fixed](https://github.com/ydb-platform/ydb/pull/13948) consistency issues related to caching deleted ranges.
```

## Version 24.3 {#24-3}

> ### Version 24.3.15.4

Reviewer 2: не смог найти это на ферме. смотрел по ссылке в pr, но как найти файл в оглавлении - не понял
если будет прямо ссылка на опубликованные доки на ферме в пеаре, где читать, будет сказочно!

Author: что такое ферма?
